File: server/python/app/application/domain/payments/payment_details/payment_product_840_specific_output.py
from dataclasses import dataclass
from typing import Optional

from app.application.domain.payments.payment_details.address import Address
from app.application.domain.payments.payment_details.payment_product_840_customer_account import PaymentProduct840CustomerAccount
from app.application.domain.payments.payment_details.protection_eligibility import ProtectionEligibility

@dataclass
class PaymentProduct840SpecificOutput:
    billing_address: Optional[Address] = None
    customer_account: Optional[PaymentProduct840CustomerAccount] = None
    customer_address: Optional[Address] = None
    protection_eligibility: Optional[ProtectionEligibility] = None

File: server/ruby/app/business/domain/payments/payment_details/payment_product840_specific_output.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class PaymentProduct840SpecificOutput
          attr_accessor :billing_address, :customer_account, :customer_address, :protection_eligibility

          def initialize(billing_address: nil, customer_account: nil, customer_address: nil, protection_eligibility: nil)
            @billing_address = billing_address
            @customer_account = customer_account
            @customer_address = customer_address
            @protection_eligibility = protection_eligibility
          end

          def to_h
            {
              'billingAddress' => @billing_address&.to_h,
              'customerAccount' => @customer_account&.to_h,
              'customerAddress' => @customer_address&.to_h,
              'protectionEligibility' => @protection_eligibility&.to_h
            }
          end
        end
      end
    end
  end
end